keps/sig-node/4205-psi-metric/README.md

```diff
@@ -0,0 +1,637 @@
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+#### Phase 1: Alpha
+
+- PSI integrated in kubelet behind a feature flag.
+- Initial e2e tests completed and enabled.
+
+#### Phase 2: Alpha
+
+- Implement Phase 2 of the enhancement which enables kubelet to 
+report node conditions based off PSI values.
+- Additional tests coverage.
+
+#### Beta
+
+- Feature gate is enabled by default.
+- Add documentation for the feature.
+- Extend e2e test coverage.
+- Allowing time for feedback.
+
+#### GA
+- TBD
+
+#### Deprecation
```

Comment:
Would this eventually deprecate disklressure and memory conditions?

Reply:
No, so the disk pressure and memory conditions are based on capacity, while the conditions in this kep are based on resource contention pressure. Maybe I can add this to the KEP and eventually in the feature doc for more clarity.

Reply:
I think it would be good to mention in the KEP the existing disk pressure and memory conditions and explain how these new conditions differ from the existing ones